modules/auth/controller.js

```diff
@@ -1,34 +1,57 @@
-const Joi = require('joi');
-const validateUser = require('./validateUser.js')
-const validatePassword = require('./validatePassword.js')
-const encryption = require('./bcryptUtils.js')
 
+const { hash } = require('bcrypt');
+const { encryption, existPassword } = require('./bcryptUtils.js')
+const { validateRegister, userExist } = require('./validation.js')
+const { usersDatabase } = require('./repository.js')
+const jwt = require('jsonwebtoken');
 
-function postLogin(req, res) {
-    res.sendStatus(200)
-}
+async function postLogin(req, res) {
+
+    try {
+
+        const validatedPayload = await validateRegister({ user: req.body.user });
 
-function postRegister(req, res) {
-    const user = req.body.user;
-    const password = req.body.password;
+        const match = await existPassword(req.body.password, req.body.user, usersDatabase);
 
-    const verifyPassword = new validatePassword(password, Joi);
-    const verifyUser = new validateUser(user, Joi);
 
-    if (password !== verifyPassword.validate()) {
-        res.send(verifyPassword.validate())
+        if (!match) {
```

Comment:
you have split responsability, you have a function "existPassword" that should be responsable if password exits but here you keep doing validation